src/log_surgeon/finite_automata/RegexAST.hpp

```diff
@@ -566,6 +652,15 @@ void RegexASTLiteral<NFAStateType>::add_to_nfa(RegexNFA<NFAStateType>* nfa, NFAS
     nfa->add_root_interval(Interval(m_character, m_character), end_state);
 }
 
+template <typename NFAStateType>
+auto RegexASTLiteral<NFAStateType>::serialize(bool const with_tags) -> std::string {
+    std::string serialized_string = std::string(1, static_cast<char>(m_character));
+    if (with_tags) {
+        serialized_string += this->serialize_negative_tags();
```

Comment:
```suggestion
        serialized_string += serialize_negative_tags();
```
Don't need `this->`. Same for the rest

Reply:
Changed.

Reply:
sorry mb. Alternative is to use `RegexAST<NFAStateType>::serialize_negative_tags()`, but might not be worth comparing to use `this`

Reply:
I think it makes sense for it to be clear that it is calling the parent method. I'll swap to being explicit.

Reply:
Changed.